source/validate_id.cpp

```diff
@@ -181,6 +181,27 @@ bool idUsage::isValid<SpvOpMemberDecorate>(const spv_instruction_t* inst,
 }
 
 template <>
+bool idUsage::isValid<SpvOpDecorationGroup>(const spv_instruction_t* inst,
+                                     const spv_opcode_desc) {
+  auto decorationGroupIndex = 1;
+  auto decorationGroup = module_.FindDef(inst->words[decorationGroupIndex]);
+
+  for (auto pair : decorationGroup->uses()) {
+    auto use = pair.first;
+    if (use->opcode() != SpvOpDecorate &&
```

Comment:
An alternative way to write this is:


 switch (use->opcode()) {
  case SpvOpDecorate:
  case SpvOpGroupDecorate:
  case SpvOpGroupMemberDecorate:
  case SpvOpName:
      break;
  default:
      ... issue the error

 }

Reply:
I tend to like that better because it evaluates the opcode only once, and it's easier to scan the list of opcodes and the logic.